build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/declarativeTask.ts

```diff
@@ -0,0 +1,80 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import globby from "globby";
+
+import type { BuildContext } from "../../buildContext";
+import type { BuildPackage } from "../../buildGraph";
+import type { DeclarativeTask } from "../../fluidBuildConfig";
+import type { TaskHandlerFunction } from "../taskHandlers";
+import { LeafTask, LeafWithFileStatDoneFileTask } from "./leafTask";
+
+class DeclarativeTaskHandler extends LeafWithFileStatDoneFileTask {
+	constructor(
+		node: BuildPackage,
+		command: string,
+		context: BuildContext,
+		taskName: string | undefined,
+		private readonly taskDefinition: DeclarativeTask,
+	) {
+		super(node, command, context, taskName);
+	}
+
+	/**
+	 * Use hashes instead of modified times in donefile.
+	 */
+	protected get useHashes(): boolean {
+		return true;
+	}
+
+	protected async getInputFiles(): Promise<string[]> {
+		const { inputGlobs, gitignore: gitignoreSetting } = this.taskDefinition;
+
+		// Ignore gitignored files if the setting is undefined, since the default is ["input"]. Otherwise check that it
+		// includes "input".
+		const gitignore: boolean =
+			gitignoreSetting === undefined || gitignoreSetting.indexOf("input") !== -1;
+		const inputFiles = await globby(inputGlobs, {
+			cwd: this.node.pkg.directory,
+			// file paths returned from getInputFiles and getOutputFiles should always be absolute
+			absolute: true,
+			gitignore,
+		});
+		return inputFiles;
+	}
+
+	protected async getOutputFiles(): Promise<string[]> {
+		const { outputGlobs, gitignore: gitignoreSetting } = this.taskDefinition;
+
+		const gitignore: boolean = gitignoreSetting?.indexOf("output") !== -1;
```

Comment:
I refactored this so it "realizes" the config (merges it with the default value), then the subsequent logic is all the
same. I was trying to "cheat" since there are only two values, but it was confusing. Should be much clearer now.